### ISSUE TITLE
CheckerFramework's @Nullable

### ISSUE THREAD

**Issue by ben-manes** [uber/NullAway#170]
Migrating from JSR-305 to the CheckerFramework and ErrorProne annotations resulted in false positives. Is `org.checkerframework.checker.nullness.qual.Nullable` supported? It doesn't appear to be.

**Comment by msridhar** [comment]
I thought we supported it.  Do you have a small example showing the problem?

**Comment by ben-manes** [comment]
I received two false positives, so perhaps there is a difference in handling? I haven't checked in my changes yet.

```
StripedBuffer.java:100: error: [NullAway] @NonNull field table not initialized
  transient volatile @Nullable Buffer<E>[] table;
```
[Here](https://github.com/ben-manes/caffeine/blob/master/caffeine/src/main/java/com/github/benmanes/caffeine/cache/StripedBuffer.java#L103) its clearly wrong.

```
JCacheLoaderAdapter.java:72: error: [NullAway] method returns @Nullable, but superclass method com.github.benmanes.caffeine.cache.CacheLoader.load(K) returns @NonNull
  public @Nullable Expirable<V> load(K key) {
```
The [superclass](https://github.com/ben-manes/caffeine/blob/master/caffeine/src/main/java/com/github/benmanes/caffeine/cache/CacheLoader.java#L62) also declares the method nullable.

**Comment by msridhar** [comment]
[Here](https://github.com/uber/NullAway/blob/master/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayNegativeCases.java#L669) are the unit tests we have around this, which are passing and haven't changed for a while.  I tried adding `transient` to the field and it still passed.  If you can give an example or push a branch showing the problem that would be helpful.

**Comment by ben-manes** [comment]
For reference, which I switch to `NullableDecl` on a field, e.g. any in Caffeine.java, I get the error,
```
Caffeine.java:168: error: [NullAway] initializer method does not guarantee @NonNull
field removalListener is initialized along all control-flow paths (remember to check for
exceptions or early returns).
  private Caffeine() {}
```
In that case the `Nullable` works fine. So odd inconsistencies.

**Comment by ben-manes** [comment]
Thanks! That fixed that scenario. The type annotation syntax will take a bit to get used to.

Any idea on the other case, where it overloads a method but both are `@Nullable`?

**Comment by msridhar** [comment]
Whoops, forgot the overload issue.  I'll have to check it tomorrow.

**Comment by msridhar** [comment]
@ben-manes I'm trying to debug the overloading issue but for some reason while running the gradle build, the "CONFIGURING" stage takes multiple minutes, with no usable output showing what is going on even with `--debug`.  Any idea what is going on?

**Comment by ben-manes** [comment]
oh, also you might want to delete anything e.g. in `.gradle`. Just a `git clean -dfx` should wipe out any weirdness. I've seen gradle act weird when versions don't match up in its `/<project>/.gradle` directory

**Comment by msridhar** [comment]
`git clean -fdx` didn't seem to fix the problem.  I had no problem yesterday building from home, so I wonder if our corporate network is causing an issue?  Here is the build scan result:

https://scans.gradle.com/s/52mz3qijiq3zs/performance/build

Looks like something inside `build.gradle` itself is taking a long time.  Anyway, I'll try again when I get home and see if that fixes the problem.

**Comment by ben-manes** [comment]
really weird. Can you try `--offline`? Your build is probably up-to-date otherwise that it will work.

Thanks again for all the help and patience 😄

**Comment by ben-manes** [comment]
np. Only other thing I can think of is to shutdown any idle daemons, in case their in a bad state (`gradle status --stop`).

**Comment by msridhar** [comment]
Because I'm obsessive, I spent more time on this and found that commenting out [this `buildScanRecipes` block](https://github.com/ben-manes/caffeine/blob/master/build.gradle#L29) from `build.gradle` fixes the slowness.  Still no idea why...

**Comment by ben-manes** [comment]
Oh, interesting! Does `--no-scan` help as an alternative?

**Comment by ben-manes** [comment]
weird. Github has been timing out today. I wonder if maybe the plugin's calls with `baseUrl` caused your build to get in a bad state. The plugin is by a Gradle developer, but I don't see his http calls using timeouts.

**Comment by msridhar** [comment]
yeah, timeouts + some logging would be good :smile:

**Comment by ben-manes** [comment]
https://github.com/melix/gradle-buildscan-recipes/issues/7

**Comment by msridhar** [comment]
As an update, I've reproduced the overriding issue on the [issue-170 branch](https://github.com/uber/NullAway/tree/issue-170).  It's a nasty problem where javac represents type use annotations parsed from bytecode differently from when the source is present.  Will take a bit of time to fix properly, but I'll try to get to it soonish.

**Comment by ben-manes** [comment]
Thanks! I'll suppress it for now.
